seatunnel-connectors-v2/connector-paimon/src/main/java/org/apache/seatunnel/connectors/seatunnel/paimon/catalog/PaimonCatalog.java

```diff
@@ -271,6 +273,20 @@ private CatalogTable toCatalogTable(
                 });
 
         List<String> partitionKeys = schema.partitionKeys();
+        List<String> primaryKyes = schema.primaryKeys();
+        if (!primaryKyes.isEmpty()) {
+            builder.primaryKey(PrimaryKey.of("pk", primaryKyes));
+            List<ConstraintKey.ConstraintKeyColumn> constraintKeyColumns =
+                    primaryKyes.stream()
+                            .map(
+                                    pk ->
+                                            new ConstraintKey.ConstraintKeyColumn(
+                                                    pk, ConstraintKey.ColumnSortType.ASC))
+                            .collect(Collectors.toList());
+            builder.constraintKey(
+                    ConstraintKey.of(
+                            ConstraintKey.ConstraintType.UNIQUE_KEY, "uk", constraintKeyColumns));
```

Comment:
I think if a field is a primary key, it must be unique. So do not put it into constraintKey.